plugins/event-bus/inmemory/src/org/apache/cloudstack/mom/inmemory/InMemoryEventBus.java

```diff
@@ -58,34 +58,6 @@ public void setName(String name) {
     }
 
     @Override
-    public UUID subscribe(EventTopic topic, EventSubscriber subscriber) throws EventBusException {
-        if (subscriber == null || topic == null) {
-            throw new EventBusException("Invalid EventSubscriber/EventTopic object passed.");
-        }
-        UUID subscriberId = UUID.randomUUID();
-
-        subscribers.put(subscriberId, new Pair<EventTopic, EventSubscriber>(topic, subscriber));
-        return subscriberId;
-    }
-
-    @Override
-    public void unsubscribe(UUID subscriberId, EventSubscriber subscriber) throws EventBusException {
-        if (subscriberId == null) {
-            throw new EventBusException("Cannot unregister a null subscriberId.");
-        }
-
-        if (subscribers.isEmpty()) {
-            throw new EventBusException("There are no registered subscribers to unregister.");
-        }
-
-        if (!subscribers.containsKey(subscriberId)) {
-            throw new EventBusException("No subscriber found with subscriber id " + subscriberId);
-        } else {
-            subscribers.remove(subscriberId);
-        }
-    }
-
-    @Override
```

Comment:
@pedro-martins Are you sure that this not used anywhere? The publish() method relies that 'subscribers' is not empty. And 'subscribers' is populated only in subscribe(). So as per the code I am sure it is getting used but not directly invoked.